File: sources/org/neochess/core/BoardHistorySlot.php
<?php

namespace org\neochess\core;

class BoardHistorySlot
{
    private $move;
    private $movingPiece;
    private $capturedPiece;
    private $epSquare;
    private $castleState;
    
    public function __construct(Move $move)
    {
        $this->move = $move;
    }
    
    public function getMovingPiece()
    {
        return $this->movingPiece;
    }

    public function setMovingPiece($movingPiece)
    {
        $this->movingPiece = $movingPiece;
    }

    public function getMove()
    {
        return $this->move;
    }

    public function getCapturedPiece()
    {
        return $this->capturedPiece;
    }

    public function getEpSquare()
    {
        return $this->epSquare;
    }

    public function getCastleState()
    {
        return $this->castleState;
    }

    public function setMove(Move $move)
    {
        $this->move = $move;
    }

    public function setCapturedPiece($capturedPiece)
    {
        $this->capturedPiece = $capturedPiece;
    }

    public function setEpSquare($epSquare)
    {
        $this->epSquare = $epSquare;
    }

    public function setCastleState($castleState)
    {
        $this->castleState = $castleState;
    }
}

File: src/main/java/org/neochess/core/BoardHistorySlot.java

package org.neochess.core;

public class BoardHistorySlot {

    private Move move;
    private Piece movingPiece;
    private Piece capturedPiece;
    private Square epSquare;
    private int castleState;

    public Move getMove() {
        return move;
    }

    public void setMove(Move move) {
        this.move = move;
    }

    public Piece getMovingPiece() {
        return movingPiece;
    }

    public void setMovingPiece(Piece movingPiece) {
        this.movingPiece = movingPiece;
    }

    public Piece getCapturedPiece() {
        return capturedPiece;
    }

    public void setCapturedPiece(Piece capturedPiece) {
        this.capturedPiece = capturedPiece;
    }

    public Square getEpSquare() {
        return epSquare;
    }

    public void setEpSquare(Square epSquare) {
        this.epSquare = epSquare;
    }

    public int getCastleState() {
        return castleState;
    }

    public void setCastleState(int castleState) {
        this.castleState = castleState;
    }
}
